### PR TITLE
fix: fixed the step interval which was being perculated to list view

### LINKED ISSUE #6261: Bug in traces query where in list view, different time has different results. 3 days result is not viewable in 10 days result.
## Bug description

*Please describe.*  
*If this affects the front-end, screenshots would be of great help.*  

## How to reproduce

1.
2.
3.

## Version information 
* **Signoz version**:
* **Browser version**:
* **Your OS and version**:
* **Your CPU Architecture**(ARM/Intel):

### PR DESCRIPTION
**Pull request by eKuG**
### Summary

Fixes : The problem where the end time was not consistent in the list view when querying clickhouse for fetching results. This was raised by a customer wherein there was a difference in results in different time scenarios.

Fixes #6261

Refer : https://signoz-team.slack.com/archives/C02C7P6KE06/p1728698795848819

#### Screenshots

NA

#### Affected Areas and Manually Tested Areas

----

> [!IMPORTANT]
> Fixes step interval calculation in `PrepareTracesQuery()` to ensure consistent end time in list view queries, with updates to test cases and Docker configurations.
> 
>   - **Behavior**:
>     - Fixes step interval calculation in `PrepareTracesQuery()` in `query_builder.go` by introducing `queryStart` and `queryEnd` to hold adjusted values.
>     - Ensures original `start` and `end` values remain unchanged for other logic.
>   - **Functions**:
>     - Updates `PrepareTracesQuery()` to use `queryStart` and `queryEnd` in calls to `buildTracesQuery()`.
>     - Affects logic for `FirstQueryGraphLimit` and `SecondQueryGraphLimit` cases.
>   - **Tests**:
>     - Updates test cases in `querier_test.go` and `v2/querier_test.go` to reflect changes in step interval calculation.
>   - **Docker**:
>     - Updates Docker image versions in `docker-compose.yaml`, `docker-compose-core.yaml`, and `docker-compose-minimal.yaml`.
>     - Removes unused environment variable `DOCKER_MULTI_NODE_CLUSTER` from `otel-collector-config.yaml`.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=SigNoz%2Fsignoz&utm_source=github&utm_medium=referral)<sup> for c4dad9fed02750d6db641c5b405edba4f05b2e21. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: pkg/query-service/app/traces/v3/query_builder.go [hunk 1]
```diff
@@ -501,19 +501,19 @@ func addOffsetToQuery(query string, offset uint64) string {
 // step is in seconds
 func PrepareTracesQuery(start, end int64, panelType v3.PanelType, mq *v3.BuilderQuery, options v3.QBOptions) (string, error) {
 	// adjust the start and end time to the step interval
-	start = start - (start % (mq.StepInterval * 1000))
-	end = end - (end % (mq.StepInterval * 1000))
+	startForGraphQuery := start - (start % (mq.StepInterval * 1000))
```
Comment (srikanthccv) [inline pkg/query-service/app/traces/v3/query_builder.go:504]: The change is only handling the graph with limits. The graph requests without limits are not handled. Alternatively, you could make this step skip for panel type List/Trace.

**Comment by eKuG** [comment]
@srikanthccv have made the changes you requested. Please review

File: pkg/query-service/app/traces/v3/query_builder.go [hunk 2]
```diff
@@ -501,26 +501,35 @@ func addOffsetToQuery(query string, offset uint64) string {
...
+	var queryStart, queryEnd int64
+
+	if panelType == v3.PanelTypeValue || panelType == v3.PanelTypeList || panelType == v3.PanelTypeTable {
+		queryStart = start
+		queryEnd = end
+	} else {
+		// adjust the start and end time to the step interval for other panel types
+		queryStart = start - (start % (mq.StepInterval * 1000))
+		queryEnd = end - (end % (mq.StepInterval * 1000))
+	}
```
Comment (srikanthccv) [inline pkg/query-service/app/traces/v3/query_builder.go:513]: I take back my previous comment. It's just cleaner to do something like this.
```go
if panelType == v3.PanelTypeGraph {
	// adjust the start and end time to the step interval for other panel types
	start = start - (start % (mq.StepInterval * 1000))
	end = end - (end % (mq.StepInterval * 1000))
}
```

Comment (eKuG) [reply pkg/query-service/app/traces/v3/query_builder.go:513]: I'll make the changes and revert back to you

Comment (eKuG) [reply pkg/query-service/app/traces/v3/query_builder.go]: @srikanthccv I have made the changes, can you review?

**Pull request merged**